Library/Homebrew/cmd/readall.rb

```diff
@@ -8,6 +8,25 @@
 
 module Homebrew
   def readall
+    if ARGV.delete("--syntax")
+      ruby_files = Queue.new
+      Dir.glob("#{HOMEBREW_LIBRARY}/{Homebrew,Formula}/**/*.rb").each do |rb|
```

Comment:
I think we can skip syntax check for formulae. It will be done in `Formulary.factory(n)` at the below.